pkg/argumentor/argumentor_interface.go

```diff
@@ -0,0 +1,62 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package argumentor
+
+import "fmt"
+
+// ArgsBuilder defines the requirements to build and manage a set of args.
+type ArgsBuilder interface {
+	Merge(Args) ArgsBuilder
+	Set(string, string) ArgsBuilder
+	Args() []string
+}
+
+// Key represents an arg key.
+type Key = string
+
+// Value represents an arg value.
+type Value = string
+
+// Args represents a set of args.
+type Args map[Key]Value
+
+// Merge implements the ArgsBuilder interface.
+func (a Args) Merge(args Args) ArgsBuilder {
+	for key, val := range args {
+		a[key] = val
+	}
+
+	return a
+}
+
+// Set implements the ArgsBuilder interface.
+func (a Args) Set(k, v Key) ArgsBuilder {
+	a[k] = v
+
+	return a
+}
+
+// Args implements the ArgsBuilder interface.
+func (a Args) Args() []string {
+	args := []string{}
+
+	for key, val := range a {
+		args = append(args, fmt.Sprintf("--%s=%s", key, val))
+	}
+
+	return args
+}
+
+// Get returns an args value.
+func (a Args) Get(k Key) Value {
+	return a[k]
```

Comment:
do we need to do this?
```
if v, ok := a[k]; ok {
  return v
}
```

Reply:
No, it will return the zero value for the map element type.